packages/x-data-grid/src/components/quickFilter/QuickFilter.tsx

```diff
@@ -78,6 +134,28 @@ function QuickFilter(props: QuickFilterProps) {
     }
   }, [quickFilterValues, formatter, parser]);
 
+  const isFirstRender = React.useRef(true);
+  const previousExpandedValue = React.useRef(expandedValue);
+  React.useEffect(() => {
+    if (isFirstRender.current) {
+      isFirstRender.current = false;
+      return;
+    }
+
+    // Ensure the expanded state has actually changed before focusing
+    if (previousExpandedValue.current !== expandedValue) {
```

Comment:
The blur event messes with the scroll position, e.g. when clicking outside of the component after the rest of the page has been scrolled:

https://github.com/user-attachments/assets/2dd47642-4d22-46fd-90ca-326d4026344e

Reply:
Fixed https://github.com/mui/mui-x/pull/16862/commits/28babf81740b57fcb859452408cc1ad27022682d